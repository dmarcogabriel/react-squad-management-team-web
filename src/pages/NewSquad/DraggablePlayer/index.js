import React from 'react';
import { useDrag } from 'react-dnd';
import cn from 'classnames';

const DraggablePlayer = ({ player }) => {
  const [{ isDragging }, drag] = useDrag({
    item: { type: 'player', id: player.player_id },
    collect: (monitor) => ({
      isDragging: !!monitor.isDragging(),
    }),
  });

  return (
    <div
      ref={drag}
      className={cn(
        'bg-gradient-to-t from-gray-400 to-white',
        'border-dashed border-2 rounded-sm p-3 my-2 cursor-move',
        isDragging && 'opacity-50'
      )}
    >
      <div className="flex flex-col md:flex-row justify-between">
        <p className="text-base font-bold">
          Name:
          <span className="text-primary-dark">{` ${player.fullname}`}</span>
        </p>
        <p className="text-base font-bold">
          Age:
          <span className="text-primary-dark">{` ${player.age}`}</span>
        </p>
      </div>
      <p className="text-base font-bold">
        Nacionality:
        <span className="text-primary-dark">{` ${player.nationality}`}</span>
      </p>
    </div>
  );
};

export default DraggablePlayer;
