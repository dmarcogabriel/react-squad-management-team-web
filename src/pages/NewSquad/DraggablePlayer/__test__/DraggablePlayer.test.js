import React from 'react';
import ReactDOM from 'react-dom';
import { DndProvider } from 'react-dnd';
import { HTML5Backend } from 'react-dnd-html5-backend';
import renderer from 'react-test-renderer';
import DraggablePlayer from '..';

const mockPlayer = {
  player_id: 'm1',
  fullname: 'Mock Player',
  age: 24,
  nationality: 'Mockland',
};

const Component = (props) => (
  <DndProvider backend={HTML5Backend}>
    <DraggablePlayer {...props} />
  </DndProvider>
);

describe('<DraggablePlayer />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<Component player={mockPlayer} />, div);
  });

  it('match snapshot', () => {
    expect(
      renderer.create(<Component player={mockPlayer} />).toJSON()
    ).toMatchSnapshot();
  });
});
