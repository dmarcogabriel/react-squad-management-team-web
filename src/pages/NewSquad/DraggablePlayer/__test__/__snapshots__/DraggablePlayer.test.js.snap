// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DraggablePlayer /> match snapshot 1`] = `
<div
  className="bg-gradient-to-t from-gray-400 to-white border-dashed border-2 rounded-sm p-3 my-2 cursor-move"
>
  <div
    className="flex flex-col md:flex-row justify-between"
  >
    <p
      className="text-base font-bold"
    >
      Name:
      <span
        className="text-primary-dark"
      >
         Mock Player
      </span>
    </p>
    <p
      className="text-base font-bold"
    >
      Age:
      <span
        className="text-primary-dark"
      >
         24
      </span>
    </p>
  </div>
  <p
    className="text-base font-bold"
  >
    Nacionality:
    <span
      className="text-primary-dark"
    >
       Mockland
    </span>
  </p>
</div>
`;
